lib/managers/KalturaFetchManager.js

```diff
@@ -287,12 +289,20 @@ class KalturaFetchManager extends kaltura.KalturaManager
 										let size = KalturaUtils.getFileSizeInBytes(filePath);
 										if (!size)
 											reject(`Fail to download ad from ${adFileUrl} to path ${filePath} with size of ${size}`);
-										This._transcodeAdAndSaveToDisk(filePath, flavorId, partnerId,
-											function (outPath, adDurationInSeconds)
+
+										TranscodingHelper.transcodeExistingFileToDisk(flavorId, This.apiConnector, partnerId, filePath, filePath, null, false).then(
+											function (transcodedFileInfo)
 											{
-												KalturaLogger.log(`Transcoded Ad to path : ${outPath}`);
-												This._updateCacheWithTranscodedAdData(adsReadyCacheKey, flavorId, adIdx, Math.floor(adDurationInSeconds * 1000), outPath, trackingList);
+												KalturaLogger.log(`Transcoded Ad to path : ${transcodedFileInfo.filePath}`);
+												This._updateCacheWithTranscodedAdData(adsReadyCacheKey, flavorId, adIdx, Math.floor(transcodedFileInfo.durationInSeconds * 1000), transcodedFileInfo.filePath, trackingList);
 												resolve();
+											},
+											function (err)
+											{
+												if(err == 'File is currently transcoding on local disk')
+													resolve();//todo?? - maybe file is processing
```

Comment:
compare to constant , add Logger about the fact that the file is currently being transcoded